### PR TITLE
Add empty-tags rule

### LINKED ISSUE #440: [Rule] Empty [Tags]
We do not cover empty [Tags]:
```
*** Test Cases ***
Test
    [Tags]
    Log  1

```

If there is empty [Tags] and:
  - there is no Default Tags that are being overwritten
  - there is Default Tags but all test cases have [Tags] (rendering Default Tags useless)
report an issue. 

There could be issue if ``__init__.robot`` contain Default Tags and we're overwriting it in child robot suite.

### PR DESCRIPTION
**Pull request by bhirsz**
Closes #440 

The rule itself is simple - the difficult part is deciding on best warning message ;)

### CODE REVIEW

File: robocop/checkers/tags.py [hunk 1]
```diff
@@ -100,6 +100,11 @@ class TagScopeChecker(VisitorChecker):
             "Tags defined in Default Tags are always overwritten",
             RuleSeverity.INFO,
         ),
+        "0608": (
+            "empty-tags",
+            "Empty [Tags]. If you are overwriting Default Tags setting use more explicit NONE",
```
Comment (mnojek) [inline robocop/checkers/tags.py:105]:
```suggestion
            "[Tags] setting without values. Consider using NONE if you want to overwrite the Default Tags",
```

Comment (mnojek) [reply robocop/checkers/tags.py:105]: Maybe this will be better?

File: tests/atest/rules/tags/empty-tags/default_and_empty_tags.robot [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    Keyword
+    One More
+
+Test With Tags
+    [Tags]    dummy
+    Log    2
+
+
+*** Keywords ***
+Keyword
```
Comment (mnojek) [inline tests/atest/rules/tags/empty-tags/default_and_empty_tags.robot:19]: Extend tests for keywords: http://robotframework.org/robotframework/latest/RobotFrameworkUserGuide.html#user-keyword-tags

**Review by mnojek — CHANGES_REQUESTED** [review]

File: tests/atest/rules/tags/empty-tags/default_and_empty_tags.robot (hunk 2 above)
Comment (bhirsz) [reply tests/atest/rules/tags/empty-tags/default_and_empty_tags.robot:19]: Sure. But " but possible Force Tags and Default Tags setting do not affect them" so I will probably put "Consider using NONE.." part only for tags from test cases.

**Review by mnojek — APPROVED** [review]

**Pull request merged**